ceph.md

```diff
@@ -219,40 +219,71 @@ to be considered secret.
 
 ## Configure Nova
 
-Use the `OpenStackDataPlane`
-[NovaTemplate](https://openstack-k8s-operators.github.io/dataplane-operator/openstack_dataplanerole/#novatemplate)
-to pass a `customServiceConfig`.
-
+Create a ConfigMap with content to be added to /etc/nova/nova.conf.d/,
+inside the nova_compute container, so that Nova uses Ceph RBD.
 ```yaml
-apiVersion: dataplane.openstack.org/v1beta1
-kind: OpenStackDataPlane
-spec:
-<snip>
-  roles:
-    edpm-compute:
-    <snip>
-        nova:
-          cellName: cell1
-          customServiceConfig: |
-            [libvirt]
-            images_type=rbd
-            images_rbd_pool=vms
-            images_rbd_ceph_conf=/etc/ceph/ceph.conf
-            images_rbd_glance_store_name=default_backend
-            images_rbd_glance_copy_poll_interval=15
-            images_rbd_glance_copy_timeout=600
-            rbd_user=openstack
-            rbd_secret_uuid=$FSID
-          deploy: true
-          novaInstance: nova
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ceph-nova
+data:
+  03-ceph-nova.conf: |
+    [libvirt]
+    images_type=rbd
+    images_rbd_pool=vms
+    images_rbd_ceph_conf=/etc/ceph/ceph.conf
+    images_rbd_glance_store_name=default_backend
+    images_rbd_glance_copy_poll_interval=15
+    images_rbd_glance_copy_timeout=600
+    rbd_user=openstack
+    rbd_secret_uuid=$FSID
 ```
+The filename, e.g. 03-ceph-nova.conf, must match "*nova*.conf" and
```

Comment:
This is interesting, I thought that an order would have been adopted within the same `Secret/ConfigMap` (e.g. I have `01-foo-nova.conf` and `03-ceph-nova.conf` within the same `ConfigMap` and I want to make sure an order is respected when processing the content of these two keys), but I didn't know this order is actually respected among multiple `ConfigMaps`.
For instance, even though I have `configmap-01` providing `03-ceph-nova.conf`, and `configmap-02` providing `01-nova-conf.conf`, I assume nova is able to process `01-nova-conf.conf` before `03-ceph-nova.conf`.
I understand that the config can be provide in a mixed form of ConfigMaps and Secrets, so it makes sense moving the logic in the nova controller.
This doesn't change anything in this doc, hence not something that should be explained here.

Reply:
> This is interesting, I thought that an order would have been adopted within the same `Secret/ConfigMap` (e.g. I have `01-foo-nova.conf` and `03-ceph-nova.conf` within the same `ConfigMap` and I want to make sure an order is respected when processing the content of these two keys), but I didn't know this order is actually respected among multiple `ConfigMaps`. For instance, even though I have `configmap-01` providing `03-ceph-nova.conf`, and `configmap-02` providing `01-nova-conf.conf`, I assume nova is able to process `01-nova-conf.conf` before `03-ceph-nova.conf`. I understand that the config can be provide in a mixed form of ConfigMaps and Secrets, so it makes sense moving the logic in the nova controller. This doesn't change anything in this doc, hence not something that should be explained here.

I was focusing about how Nova evaluates them once they land on the host. I agree we don't want to dive into the implementation details on the deployment side here.